src/read.rs

```diff
@@ -1076,6 +1083,9 @@ impl<R: Read + Seek> ZipArchive<R> {
 
     fn make_writable_dir_all<T: AsRef<Path>>(outpath: T) -> Result<(), ZipError> {
         create_dir_all(outpath.as_ref())?;
+        /* TODO: do we want to automatically make the directory writable? Wouldn't we prefer to
+         * respect the write permissions of the extraction dir? Pipelined extraction does not
+         * mutate permissions like this. */
```

Comment:
Ok, that makes sense. I was being lazy here 😅 the permissions mechanism `perms_todo` totally works to solve this (patterned after the existing `.extract()` code), but I was hoping to avoid handling perms like that. However, since this is explicitly only supporting `#[cfg(unix)]` targets for now (I'm not sure how to achieve something like `pread()` on windows), it might not be as hard as I thought. This should be easy to integrate, thanks.

Reply:
Ok, so I just added a pipelined version of that test, and it seems to work? https://github.com/zip-rs/zip2/pull/236/commits/4bbc351c6be1444b04ab35e7116418dc30a7edbe In both the current `.extract()` method and pipelined extraction, we only apply perms *after* all the files and directories are created and written. I'm leaning more towards not trying to circumvent the permissions of existing directories on disk, since I think it's very surprising that a non-writable directory would become writable just because we (e.g. accidentally) extracted a zip file into it.

Reply:
Makes sense; I agree that only new directories should be temporarily writable.